WordPress/Classes/ViewRelated/Jetpack/Login/RESTAPIJetpackLoginViewController.swift

```diff
@@ -0,0 +1,626 @@
+            }
+        }
+        .padding(.vertical, 6)
+        .padding(.horizontal, 10)
+        .background(Color(.systemBackground))
+        .cornerRadius(6)
+    }
+
+    private var statusIndicator: some View {
+        ZStack {
+            Circle()
+                .fill(indicatorBackgroundColor)
+                .frame(width: 22, height: 22)
+
+            switch stage {
+            case .pending:
+                Text(verbatim: "\(step.rawValue + 1)")
+                    .font(.system(size: 10, weight: .medium))
+                    .foregroundColor(.white)
+            case .processing:
+                ProgressView()
+                    .progressViewStyle(CircularProgressViewStyle(tint: .white))
+                    .scaleEffect(0.5)
+            case .success:
+                Image(systemName: "checkmark")
+                    .font(.system(size: 10, weight: .bold))
+                    .foregroundColor(.white)
+            case .error:
+                Image(systemName: "exclamationmark")
+                    .font(.system(size: 10, weight: .bold))
+                    .foregroundColor(.white)
+            }
+        }
+    }
+
+    private var indicatorBackgroundColor: Color {
+        switch stage {
+        case .pending:
+            return Color.gray
+        case .processing:
+            return Color.blue
+        case .success:
+            return Color.green
+        case .error:
+            return Color.red
+        }
+    }
+
+    private var borderColor: Color {
+        switch stage {
+        case .pending:
+            return Color(.systemGray5)
+        case .processing:
+            return Color.blue.opacity(0.3)
+        case .success:
+            return Color.green.opacity(0.3)
+        case .error:
+            return Color.red.opacity(0.3)
+        }
+    }
+}
+
+private enum JetpackConnectionError: LocalizedError {
+    case authenticationFailed
+    case unexpectedContext
+
+    var errorDescription: String {
+        switch self {
+        case .authenticationFailed:
+            return Strings.errorAuthenticationFailed
+        case .unexpectedContext:
+            return Strings.errorUnexpectedContext
+        }
+    }
+}
+
+private class JetpackConnectionService {
+    private let client: WordPressClient
+    private let jetpackConnectionClient: JetpackConnectionClient
+
+    init?(blog: Blog) {
+        // Requirements:
+        // - Self-hosted site, and
+        // - The site is authenticated with application password, and
+        // - Jetpack is not installed, or the installed jetpack version is 14.2 or above.
+
+        guard FeatureFlag.nativeJetpackConnection.enabled else { return nil }
+
+        guard blog.account == nil else { return nil }
+
+        guard (try? blog.getApplicationToken()) != nil else { return nil }
+
+        if let jetpack = blog.jetpack, jetpack.isInstalled, let version = jetpack.version,
+           version.compare("14.2", options: .numeric) == .orderedAscending {
```

Comment:
Please use the `Version` type for comparisons.

Reply:
The `version` here is not a strict semantic version number.

Reply:
`Version` works with any number of components from 2 to 4.

Reply:
Jetpack also has pre-release builds. I can't remember exactly what the alpha version looks like. But I'll keep this in mind, and refactor to use `Version` if applicable.